app/models/zotero_api/bibliography.rb

```diff
@@ -0,0 +1,21 @@
+module ZoteroApi
+  # A complete Zotero bibliography
+  class Bibliography < Array
+    # @return [Bibliography] all items sorted by Author-Date
+    def sort_by_author_date
+      self.class.new(sort_by(&:to_author_date))
+    end
+
+    # @param [String] `key` the bibliography item's `key` value
+    # @return [BibliographyItem] the matching item, or `nil`
+    def find(key)
+      each { |item| return item if item['key'] == key }
+      nil
+    end
+
+    # @return [String] HTML for the entire bibliography, sorted by Author-Date
+    def render
+      "<ul>#{sort_by_author_date.collect(&:to_html).join}</ul>"
```

Comment:
Ok. cool. I think I'm on the right path, and this seems much better with Rails 5 
https://medium.com/evil-martians/new-feature-in-rails-5-render-views-outside-of-actions-2fc1181e86a8#.omceel1px